Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -2060,7 +2072,7 @@ else if (!dasSnapshotFile.canRead()) {
         vmArgsRaw = "";
       }
 
-      @NonNls String serverArgsRaw = "";
+      @NonNls String serverArgsRaw = useDartdevLangServerCall ? "--protocol=analyzer" : "";
       // Note that as of Dart 2.12.0 the '--useAnalysisHighlight2' flag is ignored (and is the
       // default highlighting mode). We still want to pass it in for earlier SDKs.
       serverArgsRaw += " --useAnalysisHighlight2";
```

Comment:
It's not necessary to pass this in for the 'language-server' command.

Reply:
Thanks, updated before pushing.